### PR TITLE
persistence/genre: change Put() to upsert

### PR DESCRIPTION
**Pull request by vs49688**
Absolutely disgusting hack to work around https://github.com/navidrome/navidrome/issues/1918.
Try to insert the genre,  if it conflicts, ignore it and update the genre with the existing ID.

Can confirm this works and allows my library to scan again.

Am not expecting this to be accepted, but it's a good work-around in the meantime.

### CODE REVIEW

File: persistence/genre_repository.go [hunk 1]
```diff
@@ -36,9 +38,17 @@ func (r *genreRepository) GetAll(opt ...model.QueryOptions) (model.Genres, error
...
+		m.ID = uuid.NewString()
+	}
+
+	return r.ormer.Raw(`
+INSERT INTO genre(id, name)
+VALUES(?, ?)
+ON CONFLICT (name) DO UPDATE
+	SET name=excluded.name
+RETURNING id
+`, m.ID, m.Name).QueryRow(&m.ID)
```
Comment (vs49688) [inline persistence/genre_repository.go:51]: Probably worth mentioning - if the approach is correct, but not the raw SQL, it's easy enough to a `Get()` and then only `Put()` if it doesn't exist.

**Comment by vs49688** [comment]
Probably worth mentioning - if the approach is correct, but not the raw SQL, it's easy enough to add a `Get()` and then only `Put()` if it doesn't exist.

**Review by deluan — APPROVED** [review]
@vs49688 I've updated the branch, returned the cache and changed the upsert code a bit. Let me know if this works for you, so we can merge it.

**Comment by deluan** [comment]
Awesome! Thanks!

**Pull request merged**